src/helpers/utils.ts

```diff
@@ -35,8 +42,32 @@ export async function verify(email: string, address: string) {
   );
 }
 
-export async function unsubscribe(email: string) {
-  return await db.queryAsync('DELETE FROM subscribers WHERE email = ?', [email]);
+export async function update(email: string, address: string, subscriptions: string[]) {
+  const subs = sanitizeSubscriptions(subscriptions);
+  if (subs.length === 0) {
+    return await db.queryAsync('DELETE FROM subscribers WHERE email = ? and address = ? LIMIT 1', [
```

Comment:
another problem with completely removing email is that if a user accidentally unsubscribed from all subscription, and then want to subscribe again, he needs to pass the whole subscription flow, which is not user-friendly.

Reply:
This is by design, see discussion here: https://github.com/snapshot-labs/envelop/pull/28#issuecomment-1475966415

Reply:
I believe the `update` function should be used for subscription management in the snapshot portal,  If we are talking about the `unsubscribe` function, which can be used only inside the subscription inside the email, then probably it can remove the email completely.
If we leave it as it is, the case I have described in the second comment should be handled on the snapshot portal.

Reply:
@bonustrack what do you think about it?

Reply:
I am lost, what the request/issue ?

Reply:
My proposal is not to remove the email completely from the table during the `update` flow but remove all subscriptions instead. But the `unsubscribe` flow can remain as you did - it will remove email completely from the list.

Reply:
I am also in favor of keeping the user email on unsubscription, specially with the new preferences form: https://user-images.githubusercontent.com/7693369/238181550-d088db3a-d03a-4607-a3ea-6f57720252ef.png

By unchecking all, user may think that he can come back later to check some again, but it will not be the case.

Reply:
resolve and move discussions to https://github.com/snapshot-labs/envelop/issues/62